Macで動く？

Mac持ってないので誰か頼む。

**run時にエラー**

~~~console
❯ cargo run --bin axdl-cli
warning: /Users/goroman/Work/github/axdl-rs/axdl-gui/Cargo.toml: file `/Users/goroman/Work/github/axdl-rs/axdl-gui/src/main.rs` found to be present in multiple build targets:
  * `lib` target `axdl_gui`
  * `bin` target `axdl-gui`
   Compiling webusb-web v0.3.0
   Compiling async_zip v0.0.17
error[E0433]: failed to resolve: could not find `UsbEndpointType` in `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:302:22
    |
302 |             web_sys::UsbEndpointType::Bulk => Self::Bulk,
    |                      ^^^^^^^^^^^^^^^ could not find `UsbEndpointType` in `web_sys`

error[E0433]: failed to resolve: could not find `UsbEndpointType` in `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:303:22
    |
303 |             web_sys::UsbEndpointType::Interrupt => Self::Interrupt,
    |                      ^^^^^^^^^^^^^^^ could not find `UsbEndpointType` in `web_sys`

error[E0433]: failed to resolve: could not find `UsbEndpointType` in `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:304:22
    |
304 |             web_sys::UsbEndpointType::Isochronous => Self::Isochronous,
    |                      ^^^^^^^^^^^^^^^ could not find `UsbEndpointType` in `web_sys`

error[E0433]: failed to resolve: could not find `UsbDirection` in `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:470:22
    |
470 |             web_sys::UsbDirection::In => Self::In,
    |                      ^^^^^^^^^^^^ could not find `UsbDirection` in `web_sys`

error[E0433]: failed to resolve: could not find `UsbDirection` in `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:471:22
    |
471 |             web_sys::UsbDirection::Out => Self::Out,
    |                      ^^^^^^^^^^^^ could not find `UsbDirection` in `web_sys`

error[E0433]: failed to resolve: could not find `UsbDirection` in `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:480:42
    |
480 |             UsbDirection::In => web_sys::UsbDirection::In,
    |                                          ^^^^^^^^^^^^ could not find `UsbDirection` in `web_sys`

error[E0433]: failed to resolve: could not find `UsbDirection` in `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:481:43
    |
481 |             UsbDirection::Out => web_sys::UsbDirection::Out,
    |                                           ^^^^^^^^^^^^ could not find `UsbDirection` in `web_sys`

error[E0433]: failed to resolve: could not find `UsbDeviceFilter` in `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:558:31
    |
558 |         let filter = web_sys::UsbDeviceFilter::new();
    |                               ^^^^^^^^^^^^^^^ could not find `UsbDeviceFilter` in `web_sys`

error[E0433]: failed to resolve: could not find `UsbDeviceFilter` in `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:644:36
    |
644 |             filters.push(&web_sys::UsbDeviceFilter::from(filter));
    |                                    ^^^^^^^^^^^^^^^ could not find `UsbDeviceFilter` in `web_sys`

error[E0433]: failed to resolve: could not find `UsbDeviceRequestOptions` in `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:647:18
    |
647 |         web_sys::UsbDeviceRequestOptions::new(&filters)
    |                  ^^^^^^^^^^^^^^^^^^^^^^^ could not find `UsbDeviceRequestOptions` in `web_sys`

error[E0433]: failed to resolve: could not find `UsbTransferStatus` in `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:905:22
    |
905 |             web_sys::UsbTransferStatus::Ok => Ok(()),
    |                      ^^^^^^^^^^^^^^^^^ could not find `UsbTransferStatus` in `web_sys`

error[E0433]: failed to resolve: could not find `UsbTransferStatus` in `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:906:22
    |
906 | ...   web_sys::UsbTransferStatus::Stall => Err(Error::new(ErrorKind::Stall, "USB device sta...
    |                ^^^^^^^^^^^^^^^^^ could not find `UsbTransferStatus` in `web_sys`

error[E0433]: failed to resolve: could not find `UsbTransferStatus` in `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:907:22
    |
907 |             web_sys::UsbTransferStatus::Babble => {
    |                      ^^^^^^^^^^^^^^^^^ could not find `UsbTransferStatus` in `web_sys`

error[E0433]: failed to resolve: could not find `UsbControlTransferParameters` in `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:916:30
    |
916 |         let setup = web_sys::UsbControlTransferParameters::from(control_request);
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ could not find `UsbControlTransferParameters` in `web_sys`

error[E0433]: failed to resolve: could not find `UsbControlTransferParameters` in `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:928:30
    |
928 |         let setup = web_sys::UsbControlTransferParameters::from(control_request);
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ could not find `UsbControlTransferParameters` in `web_sys`

error[E0412]: cannot find type `UsbConfiguration` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:142:21
    |
142 | impl From<&web_sys::UsbConfiguration> for UsbConfiguration {
    |                     ^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbConfiguration` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:143:29
    |
143 |     fn from(conf: &web_sys::UsbConfiguration) -> Self {
    |                             ^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbInterface` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:147:71
    |
147 |             if let Some(iface) = iface_list.get(i).dyn_ref::<web_sys::UsbInterface>() {
    |                                                                       ^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbInterface` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:181:21
    |
181 | impl From<&web_sys::UsbInterface> for UsbInterface {
    |                     ^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbInterface` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:182:30
    |
182 |     fn from(iface: &web_sys::UsbInterface) -> Self {
    |                              ^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbAlternateInterface` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:186:67
    |
186 |             if let Some(alt) = alt_list.get(i).dyn_ref::<web_sys::UsbAlternateInterface>() {
    |                                                                   ^^^^^^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbAlternateInterface` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:229:21
    |
229 | impl From<&web_sys::UsbAlternateInterface> for UsbAlternateInterface {
    |                     ^^^^^^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbAlternateInterface` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:230:28
    |
230 |     fn from(alt: &web_sys::UsbAlternateInterface) -> Self {
    |                            ^^^^^^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbEndpoint` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:234:65
    |
234 |             if let Some(ep) = ep_list.get(i).dyn_ref::<web_sys::UsbEndpoint>() {
    |                                                                 ^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbEndpoint` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:267:21
    |
267 | impl From<&web_sys::UsbEndpoint> for UsbEndpoint {
    |                     ^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbEndpoint` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:268:27
    |
268 |     fn from(ep: &web_sys::UsbEndpoint) -> Self {
    |                           ^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbEndpointType` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:299:20
    |
299 | impl From<web_sys::UsbEndpointType> for UsbEndpointType {
    |                    ^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbEndpointType` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:300:29
    |
300 |     fn from(value: web_sys::UsbEndpointType) -> Self {
    |                             ^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbDevice` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:313:22
    |
313 |     device: web_sys::UsbDevice,
    |                      ^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbConfiguration` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:402:68
    |
402 |             if let Some(conf) = cfg_list.get(i).dyn_ref::<web_sys::UsbConfiguration>() {
    |                                                                    ^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbDevice` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:452:20
    |
452 | impl From<web_sys::UsbDevice> for UsbDevice {
    |                    ^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbDevice` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:453:30
    |
453 |     fn from(device: web_sys::UsbDevice) -> Self {
    |                              ^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbDirection` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:467:20
    |
467 | impl From<web_sys::UsbDirection> for UsbDirection {
    |                    ^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbDirection` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:468:29
    |
468 |     fn from(value: web_sys::UsbDirection) -> Self {
    |                             ^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbDirection` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:477:38
    |
477 | impl From<UsbDirection> for web_sys::UsbDirection {
    |                                      ^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbDeviceFilter` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:556:42
    |
556 | impl From<&UsbDeviceFilter> for web_sys::UsbDeviceFilter {
    |                                          ^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbRecipient` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:594:38
    |
594 | impl From<UsbRecipient> for web_sys::UsbRecipient {
    |                                      ^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbRequestType` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:616:40
    |
616 | impl From<UsbRequestType> for web_sys::UsbRequestType {
    |                                        ^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbDeviceRequestOptions` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:640:50
    |
640 | impl From<&UsbDeviceRequestOptions> for web_sys::UsbDeviceRequestOptions {
    |                                                  ^^^^^^^^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbControlTransferParameters` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:677:44
    |
677 | impl From<&UsbControlRequest> for web_sys::UsbControlTransferParameters {
    |                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `Usb` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:730:19
    |
730 |     usb: web_sys::Usb,
    |                   ^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbConnectionEvent` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:749:58
    |
749 |             Closure::wrap(Box::new(move |event: web_sys::UsbConnectionEvent| {
    |                                                          ^^^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbConnectionEvent` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:757:58
    |
757 |             Closure::wrap(Box::new(move |event: web_sys::UsbConnectionEvent| {
    |                                                          ^^^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `Usb` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:766:41
    |
766 |     fn browser_usb() -> Result<web_sys::Usb> {
    |                                         ^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbDevice` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:802:64
    |
802 |             .map(|dev| UsbDevice::from(dev.dyn_into::<web_sys::UsbDevice>().unwrap()))
    |                                                                ^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbDevice` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:812:36
    |
812 |         Ok(dev.dyn_into::<web_sys::UsbDevice>().unwrap().into())
    |                                    ^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbDevice` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:844:32
    |
844 |     fn dev(&self) -> &web_sys::UsbDevice {
    |                                ^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbTransferStatus` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:903:38
    |
903 |     fn check_status(status: web_sys::UsbTransferStatus) -> Result<()> {
    |                                      ^^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbInTransferResult` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:918:43
    |
918 |         let res = res.dyn_into::<web_sys::UsbInTransferResult>().unwrap();
    |                                           ^^^^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbOutTransferResult` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:931:43
    |
931 |         let res = res.dyn_into::<web_sys::UsbOutTransferResult>().unwrap();
    |                                           ^^^^^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbIsochronousInTransferResult` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:944:43
    |
944 |         let res = res.dyn_into::<web_sys::UsbIsochronousInTransferResult>().unwrap();
    |                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbIsochronousInTransferPacket` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:948:53
    |
948 | ..._into::<web_sys::UsbIsochronousInTransferPacket>().unwrap();
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbIsochronousOutTransferResult` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:978:43
    |
978 |         let res = res.dyn_into::<web_sys::UsbIsochronousOutTransferResult>().unwrap();
    |                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbIsochronousOutTransferPacket` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:982:53
    |
982 | ..._into::<web_sys::UsbIsochronousOutTransferPacket>().unwrap();
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbInTransferResult` in crate `web_sys`
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:996:43
    |
996 |         let res = res.dyn_into::<web_sys::UsbInTransferResult>().unwrap();
    |                                           ^^^^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0412]: cannot find type `UsbOutTransferResult` in crate `web_sys`
    --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:1010:43
     |
1010 |         let res = res.dyn_into::<web_sys::UsbOutTransferResult>().unwrap();
     |                                           ^^^^^^^^^^^^^^^^^^^^ not found in `web_sys`

error[E0277]: the trait bound `UsbDevice: From<_>` is not satisfied
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:802:24
    |
802 |             .map(|dev| UsbDevice::from(dev.dyn_into::<web_sys::UsbDevice>().unwrap()))
    |                        ^^^^^^^^^ the trait `From<_>` is not implemented for `UsbDevice`
    |
help: consider borrowing here
    |
802 |             .map(|dev| &UsbDevice::from(dev.dyn_into::<web_sys::UsbDevice>().unwrap()))
    |                        +
802 |             .map(|dev| &mut UsbDevice::from(dev.dyn_into::<web_sys::UsbDevice>().unwrap()))
    |                        ++++

error[E0599]: the method `collect` exists for struct `Map<ArrayIter<'_>, {closure@lib.rs:802:18}>`, but its trait bounds were not satisfied
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:803:14
    |
800 | /         js_sys::Array::from(&list)
801 | |             .iter()
802 | |             .map(|dev| UsbDevice::from(dev.dyn_into::<web_sys::UsbDevice>().unwrap()))
803 | |             .collect()
    | |             -^^^^^^^ method cannot be called on `Map<ArrayIter<'_>, {closure@lib.rs:802:18}>` due to unsatisfied trait bounds
    | |_____________|
    |
    |
    = note: the full type name has been written to '/Users/goroman/Work/github/axdl-rs/target/debug/deps/webusb_web-66676e9ee8c2b80c.long-type-7174072207875573485.txt'
    = note: consider using `--verbose` to print the full type name to the console
    = note: the following trait bounds were not satisfied:
            `std::iter::Map<ArrayIter<'_>, {closure@/Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:802:18: 802:23}>: Stream`
            which is required by `std::iter::Map<ArrayIter<'_>, {closure@/Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:802:18: 802:23}>: futures_util::StreamExt`
            `&std::iter::Map<ArrayIter<'_>, {closure@/Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:802:18: 802:23}>: Stream`
            which is required by `&std::iter::Map<ArrayIter<'_>, {closure@/Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:802:18: 802:23}>: futures_util::StreamExt`
            `&mut std::iter::Map<ArrayIter<'_>, {closure@/Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:802:18: 802:23}>: Stream`
            which is required by `&mut std::iter::Map<ArrayIter<'_>, {closure@/Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:802:18: 802:23}>: futures_util::StreamExt`
            `std::iter::Map<ArrayIter<'_>, {closure@/Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:802:18: 802:23}>: std::iter::Iterator`
            which is required by `&mut std::iter::Map<ArrayIter<'_>, {closure@/Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:802:18: 802:23}>: std::iter::Iterator`

error[E0282]: type annotations needed
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:394:42
    |
394 |         self.device.configuration().map(|cfg| (&cfg).into())
    |                                          ^^^         ---- type must be known at this point
    |
help: consider giving this closure parameter an explicit type
    |
394 |         self.device.configuration().map(|cfg: /* Type */| (&cfg).into())
    |                                             ++++++++++++

error[E0599]: no method named `usb` found for struct `Navigator` in the current scope
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:772:89
    |
772 | ...   Ok(usb) if !usb.is_null() && !usb.is_undefined() => return Ok(navigator.usb()),
    |                                                                               ^^^
    |
help: there is a method `sub` with a similar name, but with different arguments
   --> /private/tmp/rust-20250203-8108-he8pd4/rustc-1.84.1-src/library/core/src/ops/arith.rs:200:5

error[E0599]: no method named `usb` found for struct `WorkerNavigator` in the current scope
   --> /Users/goroman/.cargo/registry/src/index.crates.io-6f17d22bba15001f/webusb-web-0.3.0/src/lib.rs:780:89
    |
780 | ...   Ok(usb) if !usb.is_null() && !usb.is_undefined() => return Ok(navigator.usb()),
    |                                                                               ^^^
    |
help: there is a method `sub` with a similar name, but with different arguments
   --> /private/tmp/rust-20250203-8108-he8pd4/rustc-1.84.1-src/library/core/src/ops/arith.rs:200:5

Some errors have detailed explanations: E0277, E0282, E0412, E0433, E0599.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `webusb-web` (lib) due to 61 previous errors
warning: build failed, waiting for other jobs to finish...
~~~

--- a/axdl/src/communication.rs
+++ b/axdl/src/communication.rs
@@ -18,7 +18,8 @@ use std::time::Duration;
 use crate::AxdlError;
 
 const HANDSHAKE_REQUEST: [u8; 3] = [0x3c, 0x3c, 0x3c];
-pub const TIMEOUT: Duration = Duration::from_secs(5);
+pub const TIMEOUT: Duration = Duration::from_secs(10*60);
+pub const TIMEOUT_WRITE_IMAGE: Duration = TIMEOUT;
 
 pub fn wait_handshake(
     device: &mut crate::transport::DynDevice,
@@ -329,8 +330,8 @@ pub fn write_image<R: std::io::Read>(
         }
         let chunk = &buffer[..bytes_read];
         start_block(device, chunk.len() as u16)?;
-        device.write_timeout(chunk, Duration::from_secs(60))?;
-        let response = receive_response(device, Duration::from_secs(60))?;
+        device.write_timeout(chunk, TIMEOUT_WRITE_IMAGE)?;
+        let response = receive_response(device, TIMEOUT_WRITE_IMAGE)?;
         let response_view = crate::frame::AxdlFrameView::new(&response);
         if response_view.command_response() != Some(0x0080) {
             return Err(AxdlError::UnexpectedResponse(

